### PR TITLE
Re-enable Op codes for May 15 hardfork

### PR DESCRIPTION
**Pull request by ptschip**

### CODE REVIEW

**Comment by jmprcx** [comment]
I've built this locally and tested it. Looks good to me. +1

File: qa/rpc-tests/abc-monolith-activation.py [hunk 1]
```diff
@@ -49,11 +51,12 @@ def create_and_tx(self, count):
         tx.vout = []
         for _ in range(count):
             tx.vout.append(CTxOut(value, CScript([OP_1, OP_1, OP_AND])))
-        tx_signed = node.signrawtransaction(ToHex(tx))["hex"]
+        tx_signed = node.signrawtransaction(ToHex(tx),None,None,"ALL|FORKID")["hex"]
         return tx_signed
 
     def run_test(self):
         self.test = TestManager(self, self.options.tmpdir)
+        print ("num nodes is " + str(len(self.nodes)))
```
Comment (sickpig) [inline qa/rpc-tests/abc-monolith-activation.py]: I would remove this print statement before merging

File: qa/rpc-tests/abc-monolith-activation.py [hunk 2]
```diff
@@ -91,15 +94,15 @@ def spend_and():
             return tx
 
         # Check that large opreturn are not accepted yet.
-        self.log.info("Try to use the monolith opcodes before activation")
+        print("Try to use the monolith opcodes before activation")
```
Comment (sickpig) [inline qa/rpc-tests/abc-monolith-activation.py]: ditto

File: qa/rpc-tests/abc-monolith-activation.py [hunk 3]
```diff
@@ -91,15 +94,15 @@ def spend_and():
...
 
         tx0 = spend_and()
         tx0_hex = ToHex(tx0)
         assert_raises_rpc_error(-26, RPC_DISABLED_OPCODE_ERROR,
                                 node.sendrawtransaction, tx0_hex)
 
         # Push MTP forward just before activation.
-        self.log.info("Pushing MTP just before the activation and check again")
+        print("Pushing MTP just before the activation and check again")
```
Comment (sickpig) [inline qa/rpc-tests/abc-monolith-activation.py]: ditto

File: qa/rpc-tests/abc-monolith-activation.py [hunk 4]
```diff
@@ -145,24 +148,31 @@ def add_tx(block, tx):
...
         b = next_block(MONOLITH_START_TIME + 6)
         add_tx(b, tx0)
-        yield rejected(b, RejectResult(16, b'blk-bad-inputs'))
+        # In this next step we don't check the reason code for the expected failure because of timing we
+        # can not be sure whether checking the block will fail on signature validation or validation during
+        # checkinputs. We can only be certain that we must have a failure.
+        yield rejected(b)
 
-        self.log.info("Activates the new opcodes")
+        print("Activates the new opcodes")
```
Comment (sickpig) [inline qa/rpc-tests/abc-monolith-activation.py]: ditto

**Comment by sickpig** [comment]
great work @ptschip, will review later today.

File: qa/rpc-tests/abc-monolith-activation.py (hunk 1 above)
Comment (ptschip) [reply qa/rpc-tests/abc-monolith-activation.py]: done

File: qa/rpc-tests/abc-monolith-activation.py (hunk 2 above)
Comment (ptschip) [reply qa/rpc-tests/abc-monolith-activation.py]: switched to loggin.info()

File: qa/rpc-tests/abc-monolith-activation.py (hunk 3 above)
Comment (ptschip) [reply qa/rpc-tests/abc-monolith-activation.py]: switched to loggin.info()

File: qa/rpc-tests/abc-monolith-activation.py (hunk 4 above)
Comment (ptschip) [reply qa/rpc-tests/abc-monolith-activation.py]: switched to loggin.info()

**Comment by ptschip** [comment]
added -monolithactivationtime to allowed_args.  That should be the final commit other than anything else that shows up from review.

**Comment by gandrewstone** [comment]
We have been using this flag to trigger the fork:  mining.forkMay2018Time
this is redundant with "-monolithactivationtime"

Let's go with mining.forkMay2018Time since it is descriptive.
"monolith" is maybe an ABC code word?  It has no descriptive meaning, but kind of implies meaning (unlike something obviously codeword like: "BlueHawk"), so I'd like it to be removed from function calls, and file names.

Fork enabling function calls already exist but should use the date May2018 in the name.
File names could use subject based names like "opcode".

**Comment by ptschip** [comment]
@gandrewstone ok I changed monolith to may152018 for variables and filenames/testnames etc.

Also dropped the -monolithactivationtime arg and sticking with mining.forkMay15 ...

**Pull request merged**